source/features/render-conventional-commit-types.tsx

```diff
@@ -0,0 +1,82 @@
+import './render-conventional-commit-types.css';
+import React from 'react';
+import {$} from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+import observe from '../helpers/selector-observer.js';
+import {commitTitleInLists} from '../github-helpers/selectors.js';
+import {getConventionalCommitAndScopeMatch, removeCommitAndScope} from '../helpers/render-conventional-commit-types.js';
+
+const types = new Map([
+	['feat', 'Feature'],
+	['fix', 'Fix'],
+	['chore', 'Chore'],
+	['docs', 'Docs'],
+	['build', 'Build'],
+	['refactor', 'Refactor'],
+	['test', 'Test'],
+	['ci', 'CI'],
+	['perf', 'Performance'],
+]);
+
+function createLabelElement(type: string, scope?: string): JSX.Element {
+	const label = types.get(type)!;
+
+	return (
+		<span className={`IssueLabel hx_IssueLabel rgh-commit-type-label rgh-commit-type-label-${type}-colors`}>
+			{scope ? `${label}: ${scope}` : label}
+		</span>
+	);
+}
+
+function renderLabelInCommitTitle(commitTitleElement: HTMLElement): void {
+	const match = getConventionalCommitAndScopeMatch(commitTitleElement.textContent);
+
+	const {type, scope} = match?.groups ?? {};
+	if (!type || !types.has(type)) {
+		return;
+	}
+
+	removeCommitAndScope(commitTitleElement, match!);
+	commitTitleElement.prepend(createLabelElement(type, scope));
+}
+
+function renderLabelInRepoCommitTitle(commitTitleElement: HTMLElement): void {
+	const repoCommitTitleElement = $('h4 > span', commitTitleElement)!;
+	renderLabelInCommitTitle(repoCommitTitleElement);
+}
+
+function renderLabelInPRCommitTitle(commitTitleElement: HTMLElement): void {
+	const prCommitTitleElement = $('a', commitTitleElement)!;
+	renderLabelInCommitTitle(prCommitTitleElement);
+}
+
+function initRepoCommitList(signal: AbortSignal): void {
+	observe(commitTitleInLists + ' h4 > span', renderLabelInPRCommitTitle, {signal});
```

Comment:
The safe way to merge selectors is `:is(${selector}) h4 > span` as it will work even if `selector` has commas. 

One think I’d like you to check is whether both of these selectors point to a single element or if there are multiple spans/links (which I think is the case, especially if backticks and issue references appear in the commit title)

If that’s the case, the selector probably also needs `:first-child` somewhere

Reply:
My “multiple” I mean *per commit”

I know GitHub has some awkward splitting of the links so there might be multiple links and multiple spans. You can see this on any commit that has backticks or PR references (like most merged PR commits on this repo)

Reply:
> One thing I’d like you to check is whether both of these selectors point to a single element or if there are multiple spans/links (which I think is the case, especially if backticks and issue references appear in the commit title)

There is only one element matching the selector.